### DISCUSSION TITLE
오브젝트가 제페토 캐릭터 충돌했을 때 충돌감지

### DISCUSSION THREAD

**Discussion by jeejul1010** [naverz/zepeto-studio-kor#7] [5. Q&A]
OnCollisionEnter()가 동작하지 않는데요 OnControllerColliderHit()은 캐릭터가 오브젝트에 가서 부딪힐 때 충돌이 감지되는 듯 합니다. 반대로 오브젝트가 캐릭터에 충돌할 때 이벤트가 발생되게 하고 싶은데 OnTriggerEnter() 말고 혹시 방법이 있을까요? 감사합니다 :)

**Comment by moondory77-naverz** [comment]
안녕하세요. 답변이 늦어 죄송합니다.
제페토 캐릭터의 경우 Unity의 CharacterController 컴포넌트를 가지며, 이를 통해 충돌에 대한 움직임을 다루고 있습니다.
CharacterController 컴포넌트의 경우, Rigidbody가 아니기때문에 말씀주신대로 OnCollisionEnter() 이벤트가 발생하지 않는 것이 맞습니다.

말씀주시는 것과 같은 경우에는 OnTriggerEnter() 이벤트를 통해 처리주시는 것이 좋은 방법일 듯 합니다.
감사합니다.

**Comment by jeejul1010** [comment]
감사합니다!!

2021년 12월 7일 (화) 오후 8:16, moondory77-naverz ***@***.***>님이 작성:

> 안녕하세요. 답변이 늦어 죄송합니다.
> 제페토 캐릭터의 경우 Unity의 CharacterController 컴포넌트를 가지며, 이를 통해 충돌에 대한 움직임을 다루고
> 있습니다.
> CharacterController 컴포넌트의 경우, Rigidbody가 아니기때문에 말씀주신대로 OnCollisionEnter()
> 이벤트가 발생하지 않는 것이 맞습니다.
>
> 말씀주시는 것과 같은 경우에는 OnTriggerEnter() 이벤트를 통해 처리주시는 것이 좋은 방법일 듯 합니다.
> 감사합니다.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/naverz/zepeto-studio-kor/discussions/7#discussioncomment-1764395>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AI4WHUOPMSQA37JTSVZSUN3UPXUH3ANCNFSM5I4AOFSQ>
> .
> Triage notifications on the go with GitHub Mobile for iOS
> <https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675>
> or Android
> <https://play.google.com/store/apps/details?id=com.github.android&referrer=utm_campaign%3Dnotification-email%26utm_medium%3Demail%26utm_source%3Dgithub>.
>
>
-- 
hellokitty<3er
